src/Symfony/Component/Messenger/Transport/NullTransport.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+
+/**
+ * Transport that send nothing. Useful for testing purpose.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class NullTransport implements TransportInterface
+{
+    /**
+     * @var callable[]
+     */
+    private $received = array();
+
+    /**
+     * @var Envelope[]
+     */
+    private $sent = array();
+
+    /**
+     * Is the transport stopped?
+     *
+     * @var bool
+     */
+    private $stopped = false;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        if (!$this->stopped) {
+            $this->received[] = $handler;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+        $this->stopped = true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope): Envelope
+    {
+        $this->sent[] = $envelope;
+
+        return $envelope;
+    }
+
+    /**
+     * @return callable[]
+     */
+    public function getReceived(): array
```

Comment:
Is there really a use-case for this (storing and getting received handlers)?

Reply:
If you want to ensure that you received X messages yes

Reply:
But `receive` isn't called that way. Receive is [the one getting messages and calling `$handler` for each](https://github.com/symfony/symfony/blob/4bc1cc7e43e1a8e0c8c01e10da4243868bbbee2a/src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php#L43). So `receive` (usually) isn't called during a request, but [by a worker, and called only once](https://github.com/symfony/symfony/blob/4bc1cc7e43e1a8e0c8c01e10da4243868bbbee2a/src/Symfony/Component/Messenger/Worker.php#L44-L50).

Reply:
My bad, it's updated! Really look like the memory consumer though